Source/messaging/ConsoleStreamRedirect.h

```diff
@@ -0,0 +1,498 @@
+					}
+				public:
+					ResourceMonitor(ResourceMonitor&&) = delete;
+					ResourceMonitor(const ResourceMonitor&) = delete;
+					ResourceMonitor& operator= (ResourceMonitor&&) = delete;
+					ResourceMonitor& operator= (const ResourceMonitor&) = delete;
+
+					static ResourceMonitor& Instance() {
+						static ResourceMonitor instance;
+						return (instance);
+					}
+					~ResourceMonitor() override {
+						Core::Thread::Stop();
+						Core::Thread::Wait(Core::Thread::BLOCKED | Core::Thread::STOPPED, Core::infinite);
+					}
+
+				public:
+					void Register(ReaderImplementation& source) {
+						_adminLock.Lock();
+						ASSERT(std::find(_resources.begin(), _resources.end(), &source) == _resources.end());
+						_resources.push_back(&source);
+						source.Read(_event);
+						if (_resources.size() == 1) {
+							Core::Thread::Run();
+						}
+						_adminLock.Unlock();
+					}
+					void Unregister(ReaderImplementation& source) {
+						_adminLock.Lock();
+						typename Implementations::iterator entry(std::find(_resources.begin(), _resources.end(), &source));
+						ASSERT(entry != _resources.end());
+						if (entry != _resources.end()) {
+							source.Cancel(_event);
+							_resources.erase(entry);
+						}
+						if (_resources.empty() == true) {
+							Core::Thread::Block();
+							::SetEvent(_event.hEvent);
+						}
+						_adminLock.Unlock();
+					}
+
+					uint32_t Worker() override {
+						if (::WaitForSingleObject(_event.hEvent, Core::infinite) == WAIT_OBJECT_0) {
+							::ResetEvent(_event.hEvent);
+							_adminLock.Lock();
+							// Iterate through the list of entries to flush the data read:
+							for (ReaderImplementation* entry : _resources) {
+								entry->Process(_event);
+							}
+							_adminLock.Unlock();
+						}
+						return(Core::infinite);
+					}
+
+				private:
+					Core::CriticalSection _adminLock;
+					Implementations _resources;
+					OVERLAPPED _event;
+				};
+
+			public:
+				ReaderImplementation() = delete;
+				ReaderImplementation(ReaderImplementation&&) = delete;
+				ReaderImplementation(const ReaderImplementation&) = delete;
+				ReaderImplementation& operator=(ReaderImplementation&&) = delete;
+				ReaderImplementation& operator=(const ReaderImplementation&) = delete;
+
+				ReaderImplementation(ParentClass& parent)
+					: Reader(parent)
+					, _index(Core::IResource::INVALID)
+					, _copy(Core::IResource::INVALID)
+					, _handle(nullptr) {
+				}
+				~ReaderImplementation() {
+					Close();
+				}
+
+			public:
+				bool Open(const Core::IResource::handle replacing) {
+
+					ASSERT(_index == Core::IResource::INVALID);
+					ASSERT(replacing != Core::IResource::INVALID);
+
+					if (replacing != Core::IResource::INVALID) {
+						Core::IResource::handle newDescriptor;
+
+						if (CreateOverlappedPipe(_handle, newDescriptor) != false) {
+							_flushall();
+							_copy = ::_dup(replacing);
+							if (::_dup2(newDescriptor, replacing) == -1) {
```

Comment:
Yet another school day, learned about _dup, _dup2 and file descriptors in general

Reply:
Take care the official, correct Posix calls are without the underscore, this is windows crap ;-)